Starcraft2Client.py

```diff
@@ -149,30 +156,31 @@ def on_package(self, cmd: str, args: dict):
             self.difficulty = args["slot_data"]["game_difficulty"]
             self.all_in_choice = args["slot_data"]["all_in_map"]
             slot_req_table = args["slot_data"]["mission_req"]
-            self.mission_req_table = {}
-            # Compatibility for 0.3.2 server data.
-            if "category" not in next(iter(slot_req_table)):
-                for i, mission_data in enumerate(slot_req_table.values()):
-                    mission_data["category"] = wol_default_categories[i]
-            for mission in slot_req_table:
-                self.mission_req_table[mission] = MissionInfo(**slot_req_table[mission])
+            self.mission_req_table = {
+                mission: MissionInfo(**slot_req_table[mission]) for mission in slot_req_table
+            }
+
+            self.build_location_to_mission_mapping()
 
             # Look for and set SC2PATH.
             # check_game_install_path() returns True if and only if it finds + sets SC2PATH.
             if "SC2PATH" not in os.environ and check_game_install_path():
                 check_mod_install()
 
-        if cmd in {"PrintJSON"}:
-            if "receiving" in args:
-                if self.slot_concerns_self(args["receiving"]):
-                    self.announcements.append(args["data"])
-                    return
-            if "item" in args:
-                if self.slot_concerns_self(args["item"].player):
-                    self.announcements.append(args["data"])
-
+    def on_print_json(self, args: dict):
+        relevant = False
+        if "receiving" in args:
+            if self.slot_concerns_self(args["receiving"]):
+                relevant = True
+        if "item" in args:
+            if self.slot_concerns_self(args["item"].player):
+                relevant = True
+        if relevant:
+            self.announcements.append(self.raw_text_parser(copy.deepcopy(args["data"])))
```

Comment:
I don't think the deepcopy is strictly required here, because the super().on_print_json should not care that some field are pre-filled? Does not hurt though, since there is no promise that they don't interfere.

Reply:
the parser is considered destructive as it transforms the object. not deep-copying might work or might not, I didn't even try. I simply went for the deepcopy to avoid any future problems from weird interactions.